components/cdc/src/observer.rs

```diff
@@ -91,6 +103,58 @@ impl CdcObserver {
             .get(&region_id)
             .cloned()
     }
+
+    fn untrack_raw_ts(&self, raw_region_ts: Vec<RawRegionTs>) {
+        if raw_region_ts.is_empty() {
+            return;
+        }
+        if let Err(e) = self.sched.schedule(Task::RawUntrackTs { raw_region_ts }) {
+            warn!("cdc schedule task failed"; "error" => ?e);
+        }
+    }
+
+    // parse rawkv cmd from CmdBatch Vec and return the max ts of every region.
+    pub fn get_raw_region_ts(&self, cmd_batches: &Vec<CmdBatch>) -> Vec<RawRegionTs> {
+        let mut region_ts = vec![];
+        for batch in cmd_batches {
+            if batch.is_empty() {
+                continue;
+            }
+            let region_id = batch.region_id;
+            let cdc_id = batch.cdc_id;
+            if let Some(ob_id) = self.is_subscribed(region_id) {
+                if ob_id != cdc_id {
+                    continue;
+                }
+            } else {
+                continue;
+            }
+            let cmd = &batch.cmds.last().unwrap();
```

Comment:
Other items in `cmds` should also be scan, in case the last raw put entry is not in the last `cmd`.

Reply:
Done